From 2ef158a26834e47ca8a38dfc09b65da28a29526a Mon Sep 17 00:00:00 2001
From: stenkinevgeniy <stenkinevgeniy@gmail.com>
Date: Tue, 8 May 2018 06:52:05 +0000
Subject: [PATCH 19/35] Audio: out_open_pcm_devices - use out->config instead
 of pcm_device->pcm_profile->config.

Current get_pcm_device return always first finded pcm_profile.
And out->pcm_dev_list always have only pcm_device_playback pcm_profiles.

But if we need use deep_buffer pcm device
we need use deep_bufer pcm_config - current hal
don't did this...

And we configure out->config in adev_open_output_stream
function with proper pcm_config (deep or no).

Change-Id: I86fdb18202afb0e83393f32830d57c929f174ae6
(cherry picked from commit 39d1d019965d04ddfad2c643de53cd5501803aa4)
---
 audio/audio_hw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/audio/audio_hw.c b/audio/audio_hw.c
index c76bac4..ce452ef 100644
--- a/audio/audio_hw.c
+++ b/audio/audio_hw.c
@@ -2421,7 +2421,7 @@ static int out_open_pcm_devices(struct stream_out *out)
               __func__, pcm_device_card, pcm_device_id);
 
         pcm_device->pcm = pcm_open(pcm_device_card, pcm_device_id,
-                               PCM_OUT | PCM_MONOTONIC, &pcm_device->pcm_profile->config);
+                               PCM_OUT | PCM_MONOTONIC, &out->config);
 
         if (pcm_device->pcm && !pcm_is_ready(pcm_device->pcm)) {
             ALOGE("%s: %s", __func__, pcm_get_error(pcm_device->pcm));
-- 
2.17.1

